cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamUtils.java

```diff
@@ -136,6 +136,25 @@ public static int getSequenceId(String name) {
   }
 
   /**
+   * Returns the writer instance number of the given file name.
+   *
+   * @param name Name of the file.
+   * @return The writer instance number of the stream file.
+   * @see StreamInputFormat
+   */
+  public static int getWriterInstanceId(String name) {
```

Comment:
Oh, then there isn't always an instance id in the prefix? It is enforced nowhere? Is there another way to make sure to get the files that one writer has written?